docs/general/ddl/data-types.rst

```diff
@@ -247,6 +247,46 @@ are likely to be larger due to additional metadata.
          defining a column to not use the column store or by :ref:`turning off indexing
          <sql_ddl_index_off>`.
 
+Precedence and type conversion
+------------------------------
+
+When expressions of different data types are combined by operators or scalars,
+the data type with the lower precedence is converted to the data type
+with the higher precedence. If the conversion isn't a supported implicit
+conversion, an error is returned.
+
+The following precedence order is used for data types (highest to lowest):
+
+1. Custom (complex) types (currently: :ref:`bitstring <data-types-bit-strings>`,
+   :ref:`float_vector <type-float_vector>`) (highest)
+2. :ref:`GEO_SHAPE <data-types-geo-shape>`
+3. :ref:`JSON <data-type-json>`
+4. :ref:`OBJECT <data-types-objects>`
+5. :ref:`GEO_POINT <data-types-geo-point>`
+6. ``Record`` (internal type, return type of
+   :ref:`table functions <table-functions>`)
+7. :ref:`Array <data-types-arrays>`
+8. :ref:`Numeric <data-types-numeric>`
+9. :ref:`Double precision <type-double-precision>`
+10. :ref:`Real <type-real>`
+11. :ref:`IP <data-types-ip-addresses>`
+12. :ref:`Bigint <type-bigint>`
+13. :ref:`Timestamp without time zone <type-timestamp-without-tz>`
+14. :ref:`Timestamp with time zone <type-timestamp-with-tz>`
```

Comment:
I realize this is just documentation, but why is without timezone stronger than with timezone? It seems like with-tz is without-tz plus additional info

Reply:
> but why is without timezone stronger than with timezone? It seems like with-tz is without-tz plus additional info

Because a time stamp with time zone can be cast to a time stamp without time zone, but not the other way around without loosing information (the time zone).